src/lbp.jl

```diff
@@ -0,0 +1,146 @@
+function lbp_original(bit_pattern::Array{Bool, 1})
+	sum([b * 2 ^ (length(bit_pattern) - i) for (i, b) in enumerate(bit_pattern)])
+end
+
+UNIFORM_LBP_TABLE = Dict{Array{Bool, 1}, Int}()
+UNIFORM_PATTERN_COUNT = 0
+NON_UNIFORM_PATTERN = Array{Bool, 1}()
+
+function init_uniform_lbp_params(points::Integer)
+	global UNIFORM_LBP_TABLE, NON_UNIFORM_PATTERN, UNIFORM_PATTERN_COUNT
+	
+	UNIFORM_PATTERN_COUNT::Int = 0
+	empty!(UNIFORM_LBP_TABLE)
+	NON_UNIFORM_PATTERN = zeros(Bool, points)
+	NON_UNIFORM_PATTERN[1:2:points] = true
+	UNIFORM_LBP_TABLE[NON_UNIFORM_PATTERN] = points * (points - 1) + 2
+end
+
+function lbp_uniform(bit_pattern::Array{Bool, 1})
+	global UNIFORM_LBP_TABLE, NON_UNIFORM_PATTERN, UNIFORM_PATTERN_COUNT
+	variations = sum([bit_pattern[i] != bit_pattern[i + 1] for i in 1:length(bit_pattern) - 1])
```

Comment:
This will allocate an array on each call. Assuming this is a performance-critical function, much better to just write this out as a loop. Or if you want to support julia-0.5 only, use the new generator syntax, which is just exactly what you wrote but without the `[]` brackets.